akka-docs/rst/java/code/jdocs/akka/typed/IntroSpec.java

```diff
@@ -0,0 +1,229 @@
+        this.screenName = screenName;
+        this.replyTo = replyTo;
+      }
+    }
+    //#chatroom-protocol
+    //#chatroom-behavior
+    private static final class PostSessionMessage implements Command {
+      public final String screenName;
+      public final String message;
+      public PostSessionMessage(String screenName, String message) {
+        this.screenName = screenName;
+        this.message = message;
+      }
+    }
+    //#chatroom-behavior
+    //#chatroom-protocol
+
+    static interface SessionEvent {}
+    public static final class SessionGranted implements SessionEvent {
+      public final ActorRef<PostMessage> handle;
+      public SessionGranted(ActorRef<PostMessage> handle) {
+        this.handle = handle;
+      }
+    }
+    public static final class SessionDenied implements SessionEvent {
+      public final String reason;
+      public SessionDenied(String reason) {
+        this.reason = reason;
+      }
+    }
+    public static final class MessagePosted implements SessionEvent {
+      public final String screenName;
+      public final String message;
+      public MessagePosted(String screenName, String message) {
+        this.screenName = screenName;
+        this.message = message;
+      }
+    }
+
+    public static final class PostMessage {
+      public final String message;
+      public PostMessage(String message) {
+        this.message = message;
+      }
+    }
+    //#chatroom-protocol
+    //#chatroom-behavior
+
+    public static Behavior<Command> chatRoom() {
+      return chatRoom(new ArrayList<ActorRef<SessionEvent>>());
+    }
+
+    private static Behavior<Command> chatRoom(List<ActorRef<SessionEvent>> sessions) {
+      return Actor.stateful((ctx, msg) -> {
+        if (msg instanceof GetSession) {
+          GetSession getSession = (GetSession) msg;
+          ActorRef<PostMessage> wrapper = ctx.createAdapter(p ->
+            new PostSessionMessage(getSession.screenName, p.message));
+          getSession.replyTo.tell(new SessionGranted(wrapper));
+          // TODO mutable collection :(
+          List<ActorRef<SessionEvent>> newSessions = new ArrayList<ActorRef<SessionEvent>>(sessions);
+          newSessions.add(getSession.replyTo);
+          return chatRoom(newSessions);
+        } else if (msg instanceof PostSessionMessage) {
+          PostSessionMessage post = (PostSessionMessage) msg;
+          MessagePosted mp = new MessagePosted(post.screenName, post.message);
+          sessions.forEach(s -> s.tell(mp));
+          return Actor.same();
+        } else {
+          return Actor.unhandled();
+        }
+      });
+    }
+    //#chatroom-behavior
+
+    public static Behavior<Command> mutableChatRoom() {
+      return Actor.deferred(c -> {
+        final List<ActorRef<SessionEvent>> sessions = new ArrayList<ActorRef<SessionEvent>>();
+        int postCount = 0;
+
+        return Actor.stateful((ctx, msg) -> {
```

Comment:
to structure larger actors the message matches should delegate to methods, but there is no place to write such methods without an enclosing class